packages/mongodb-chatbot-server/src/test/testHelpers.ts

```diff
@@ -63,6 +76,80 @@ export async function makeTestApp(defaultConfigOverrides?: PartialAppConfig) {
   };
 }
 
+export const TEST_OPENAI_API_KEY = "test-api-key";
+
+/**
+  Helper function to quickly make a local server for testing purposes.
+  Builds on the other helpers for app/config stuff.
+  @param defaultConfigOverrides - optional overrides for default app config
+ */
+export const makeTestLocalServer = async (
+  defaultConfigOverrides?: PartialAppConfig,
+  port?: number
+) => {
+  const testAppResult = await makeTestApp({
+    ...defaultConfigOverrides,
+    port,
+  });
+
+  const server = testAppResult.app.listen(port ?? TEST_PORT);
+
+  return { ...testAppResult, server };
+};
+
+export const makeOpenAiClient = (origin: string, ipAddress: string) => {
+  return new OpenAI({
+    baseURL: origin + DEFAULT_API_PREFIX,
+    apiKey: TEST_OPENAI_API_KEY,
+    defaultHeaders: {
+      Origin: origin,
+      "X-Forwarded-For": ipAddress,
+    },
+  });
+};
+
+export const makeCreateResponseRequest = (
+  openAiClient: OpenAI,
+  body?: Partial<CreateResponseRequest["body"]>
+) => {
+  return openAiClient.responses
+    .create({
+      ...basicResponsesRequestBody,
+      ...body,
+    })
+    .withResponse();
+};
+
+export const formatOpenAIStreamError = (
+  httpStatus: number,
+  code: ERROR_CODE,
+  message: string,
+  headers?: Record<string, string>
+) => {
+  const error = new APIError(httpStatus, { code, message }, message, headers);
+  return makeOpenAIStreamError(error);
+};
+
+/**
+ Helper function to collect a full response from a stream.
+ */
+export const collectStreamingResponse = async (response: Response) => {
```

Comment:
maybe stupid question, but rather than collecting the stream, can't u just use the objects returned by the call to `openAiClient.responses.create({})`?

Reply:
maybe in addition to testing the response.text(), we can also test the stream returned by the OpenAIClient.responses.create() method? think can approach in a few ways:

1. add separate helper function `expectCreateResponse`, adding to all relevant tests
2. add logic to some of existing helpers
3. add distinct tests to validate the client functionality.

Reply:
I'll look into testing the stream though!